modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/batch/ShredJob.scala

```diff
@@ -188,18 +191,16 @@ object ShredJob {
     spark: SparkSession,
     igluConfig: Json,
     duplicateStorageConfig: Option[Json],
-    formats: Formats,
-    shredderConfig: Shredder.Batch,
-    sqsQueue: String,
-    region: String
+    config: Config[StorageTarget],
+    shredderConfig: Shredder.Batch
   ): Unit = {
     val atomicLengths = EventUtils.getAtomicLengths(IgluSingleton.get(igluConfig).resolver).fold(err => throw err, identity)
```

Comment:
This line can also throw an error.  So can many other lines of this `run` function - not just the one you have wrapped in `Either.catchNonFatal`.  It would be great if Sentry can capture any exception that gets thrown at any point during processing.

Did you consider wrapping the entire `run` block inside `Either.catchNonFatal` so you catch everything?

Reply:
sure, I will move it to wrap entire `run` execution